### PR TITLE
Update Welcome a first-time contributor example

### PR DESCRIPTION
**Pull request by calkeo**
This PR improves the pagination logic for the "Welcome a first-time contributor" example in the README.

The [current example](https://github.com/actions/github-script#welcome-a-first-time-contributor) retrieves all issues created by the given user. This means that if there's a PR that passes `issue.number !== context.issue.number && issue.pull_request` on the first page of results, the logic will still retrieve data for all remaining pages. This can add quite a lot of overhead on repositories where the given user has authored many issues.

This PR stops the pagination early by calling the `done()` method if a match is found.

I appreciate that the current example may be deliberate for the purpose of keeping the example simple and easily-readable, but I thought I'd open this up just in case.

### CODE REVIEW

**Comment by joshmgross** [comment]
Thanks @calkeo!

Were you able to test these changes in a workflow?

File: README.md [hunk 1]
```diff
@@ -166,16 +166,18 @@ jobs:
...
+                done() // Stop paginating
+                return pr
               }
+            })
 
-              if (issue.pull_request) {
-                return // Creator is already a contributor.
-              }
+            if (issues.length) {
+              return
```
Comment (joshmgross) [inline README.md:180]: Should we keep the comment here?

```suggestion
              return // Creator is already a contributor.
```

Comment (ArjunSharda) [reply README.md:180]: Tbf yes.

Comment (navarroaxel) [reply README.md:180]: Perhaps this format is better.
```suggestion
              // Creator is already a contributor.
              return
```

**Comment by joshmgross** [comment]
👋 Going to close this out, feel free to reopen if you'd like to pick this back up.

There's still a pending question: https://github.com/actions/github-script/pull/257#issuecomment-1100273964

**Pull request closed without merging**